network/src/protocols/rpc/fuzzing.rs

```diff
@@ -63,8 +64,8 @@ pub fn generate_corpus(gen: &mut ValueGenerator) -> Vec<u8> {
 
 // Fuzz the inbound rpc protocol.
 pub fn fuzzer(data: &[u8]) {
-    let network_context = NetworkContext::mock();
-    let connection_metadata = ConnectionMetadata::mock(MOCK_PEER_ID);
+    let network_context = NetworkContext::mock_with_peer_id(MOCK_PEER_ID_1);
+    let connection_metadata = ConnectionMetadata::mock(MOCK_PEER_ID_2);
```

Comment:
Interesting, so Prometheus counters blow up from having high cardinality of peer ids in the tests?

I wonder if we want that in this kind of testing in general, or to find a way to shut off the metrics system for things like fuzzing.

I think though, in general, testing with mock hardcoded values like this is probably a better strategy, unless we're trying to specifically fuzz the peer ids

Reply:
Yep, every unique {counter name, label set} adds an entry to the global metrics registry which never goes away.

I don't think this is actually a problem necessarily. The issue here is non-representative usage causing metrics blowup. Specifically, sampling a new `NetworkContext::mock()` every _iteration_ is like testing if our node software can handle running 200,000 concurrent network stacks without emitting too many metrics, which is obviously not representative of actual usage where we have a O(1) # of networks.

In contrast, there's an argument to be made that it's actually _good_ to sample different peer id's for the `ConnectionMedadata` since that would effectively assert that we don't have any metrics specifically tied to the remote peer id.